jni/include/faiss_index_create.h

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+/**
+ * This file contains util methods which are free of JNI to be used in faiss_wrapper.cpp
+ */
+
+#ifndef OPENSEARCH_KNN_FAISS_INDEX_CREATE_H
+#define OPENSEARCH_KNN_FAISS_INDEX_CREATE_H
+
+#include "jni_util.h"
+#include "faiss/impl/io.h"
+#include "faiss/index_factory.h"
+#include "faiss/index_io.h"
+#include "faiss/IndexHNSW.h"
+#include "faiss/IndexIVFFlat.h"
+#include "faiss/MetaIndexes.h"
+#include "faiss/Index.h"
+#include "faiss/impl/IDSelector.h"
+#include "faiss/IndexIVFPQ.h"
+#include "faiss/IndexBinaryIVF.h"
+#include "faiss/IndexBinaryHNSW.h"
+#include "faiss/IndexBinary.h"
+#include <algorithm>
+#include <string>
+#include <vector>
+#include <memory>
+#include <jni.h>
+
+namespace knn_jni {
+namespace faiss_wrapper {
+
+class IndexCreateService {
+    public:
+        IndexCreateService(
+            knn_jni::JNIUtilInterface * jniUtil,
+            JNIEnv * env,
+            faiss::MetricType metric,
+            std::string indexDescription,
+            int dim,
+            int numIds,
+            int threadCount,
+            int64_t vectorsAddress,
+            std::vector<int64_t> ids,
+            std::string indexPath,
+            std::unordered_map<std::string, jobject> parameters
+        );
+        virtual void create();
+        virtual ~IndexCreateService() = default;
+    protected:
+        knn_jni::JNIUtilInterface * jniUtil;
+        JNIEnv * env;
+        faiss::MetricType metric;
+        std::string indexDescription;
+        int dim;
+        int numIds;
+        int threadCount;
+        int64_t vectorsAddress;
+        std::vector<int64_t> ids;
+        std::string indexPath;
+        std::unordered_map<std::string, jobject> parameters;
+    private:
+
+};
+
+class BinaryIndexCreateService : public IndexCreateService {
+    public:
+        BinaryIndexCreateService(
+            knn_jni::JNIUtilInterface * jniUtil,
+            JNIEnv * env,
+            faiss::MetricType metric,
+            std::string indexDescription,
+            int dim,
+            int numIds,
+            int threadCount,
+            int64_t vectorsAddress,
+            std::vector<int64_t> ids,
+            std::string indexPath,
+            std::unordered_map<std::string, jobject> parameters
+        );
+        void create() override;
+};
+
+class IndexCreateServiceFactory {
```

Comment:
nit: formatting

Reply:
If you are referring the missing indentation with two namespaces, it is by intention to avoid having large space for each line.